metricbeat/module/system/memory/_meta/fields.yml

```diff
@@ -79,6 +79,29 @@
           description: >
             Available swap memory.
 
+        - name: vmstat
+          type: group
+          description: This group contains swap stats from vmstats on linux
+          fields:
+            - name: pswpin
+              type: long
+              description: Pages swapped in
+            - name: pswpout
+              type: long
+              description: Pages swapped out
+            - name: thp_swpout
+              type: long
+              description: Huge page swap stats
+            - name: thp_swpout_fallback
+              type: long
+              description: Huge page swap stats
+            - name: swap_ra
+              type: long
+              description: Swap readahead stats
+            - name: swap_ra_hit
+              type: long
+              description: Swap readahead stats
```

Comment:
I am not sure about these names, we would have duplication of "swap" in the name (e.g. `system.memory.swap.vmstat.pswpin`), and I wonder if `vmstat` provides any info here.

I'd propose to use metrics like these ones:
* `system.memory.swap.in.pages` (what would also allow us to have `system.memory.swap.in.bytes` if we know the page sizes)
* `system.memory.swap.out.pages`
* `system.memory.hugepages.swap.out.pages` (we already have a group for hugepages)
* `system.memory.hugepages.swap.in.pages`
* `system.memory.swap.readahead.pages`? (What is the unit of readahead stats?)
* `system.memory.swap.readahead.hits`?

Readahead stats would probably need a better description.

They are all suggestions :slightly_smiling_face: let me know what you think.